### DISCUSSION TITLE
Compatible device ?

### DISCUSSION THREAD

**Discussion by sara-caballero** [ikarus23/MifareClassicTool#467] [Q&A]
How can I find out which devices support MIFARE Classic? My phone is not listed among either the compatible or incompatible devices. My phone is Oppo Find X5 Pro

**Comment by ikarus23** [comment]
Hi. Sorry, most of the time manufactures do not declare if their device support MIFARE Classic or not. Typically there is only information about being NFC compatible or not -- which is not helpful.

The first step is to do what you did: research. [COMPATIBLE_DEVICES.md](https://github.com/ikarus23/MifareClassicTool/blob/master/COMPATIBLE_DEVICES.md) and [INCOMPATIBLE_DEVICES.md](https://github.com/ikarus23/MifareClassicTool/blob/master/INCOMPATIBLE_DEVICES.md) are good resources but also check out sites like https://www.shopnfc.com/en/content/7-nfc-compatibility or https://devices.nfc.help/. And of course, try some google, duckduckgo, etc.

If you are not able to find anything, you need to do practical testing your self. However, you need a reference for this. In this case a tag/card that you are 100% certain is MIFARE Classic. There is no app that can tell you for certain if MIFARE Classic is supported without a MIFARE Classic card. Once you have a card just present it to your phone running apps like [NXP's TagInfo](https://play.google.com/store/apps/details?id=com.nxp.taginfolite), [NFC Tools](https://play.google.com/store/apps/details?id=com.wakdev.wdnfc) or this app ([MCT](https://play.google.com/store/apps/details?id=de.syss.MifareClassicTool)). Don't get fooled if an app shows you information like the UID (serial number). This will work on all NFC enabled devices regardless of actual MIFARE Classic support. However, the mentioned apps should tell you right away, if they detected they can not read/write your reference MIFARE Classic card. Then you know.

Be aware! MIFARE Classic support can change! I have seen it both ways: After an update an unsupported device was able to read/write MIFARE Classic and other supported devices were not longer able to read/write MIFARE classic. Most of the time this is because of changes in drivers or firmware for the NFC chip in your phone.

Hope this helps!

**Reply by robotJie** [reply]
@ikarus23 
So, in my understanding, as long as the hardware has the corresponding capability, even if it's not an NXP NFC controller, it's technically possible to read/write MiFare cards, right?

**Reply by ikarus23** [reply]
Yes, there are other controllers out there which can read MIFARE Classic just fine. Technically speaking, a lot of other controllers can read/write MIFARE Classic. But support (e.g. driver), especially on phones, is lacking.
